kotlinx-coroutines-core/common/src/flow/operators/Delay.kt

```diff
@@ -129,7 +151,98 @@ public fun <T> Flow<T>.debounce(timeoutMillis: Long): Flow<T> {
  */
 @ExperimentalTime
 @FlowPreview
-public fun <T> Flow<T>.debounce(timeout: Duration): Flow<T> = debounce(timeout.toDelayMillis())
+public fun <T> Flow<T>.debounceWithDuration(timeout: Duration): Flow<T> = debounce(timeout.toDelayMillis())
```

Comment:
It's not ideal that I am renaming this method from `Flow<T>.debounce(timeout: Duration): Flow<T>`. 

The selector variation method name `Flow<T>.debounce(timeout: (T) -> Duration): Flow<T>` below was conflicting with `Flow<T>.debounce(timeoutMillis: (T) -> Long): Flow<T>`. I could just rename only the duration selector variation to e.g. `Flow<T>.debounceWithDurationSelector(timeout: (T) -> Duration): Flow<T>`, but I felt this would be a better grouping to keep the word `duration` in the method names for both instead of just one of them. 

Please let me know if renaming an existing method concerns.

Reply:
The right way to do it is this:
* Keep the Kotlin name `debounce`
* Add `@JvmName("debounceDuration")` to avoid platform declaration clash (we don't use `With` or other prepositions in these kind of names).

Reply:
`@JvmName` does not seem to work when a function argument is used. I see the issue mentioned in https://youtrack.jetbrains.com/issue/KT-22119. I pushed with `debounceDuration(...)`.